.github/workflows/deployment.yaml

```
@@ -0,0 +1,226 @@
+        run: |
+          set -euo pipefail
+          TEAMS_LIST=$(echo $TEAMS | tr "," "\n")
+
+          CHECK_SUCCESS=0
+
+          for TEAM in $TEAMS_LIST; do
+            set +e
+            # Check current user membership with gh api
+            echo "Check if user is member of $TEAM"
+            GH_TEAM_MEMBERSHIP_RESPONSE=$(gh api -i --method GET -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" \/orgs/$ORG/teams/$TEAM/memberships/$USER)
+            set -e
+
+            # Extract the HTTP status code
+            HTTP_CODE=$(echo $GH_TEAM_MEMBERSHIP_RESPONSE | head -n 1 | cut -d' ' -f 2)
+
+            if [[ $HTTP_CODE -eq 200 ]]; then
+              echo "User is member of $TEAM"
+              CHECK_SUCCESS=1
+              break
+            else
+              echo "User is NOT member of $TEAM"
+            fi
+          done
+
+          if [[ $CHECK_SUCCESS -eq 0 ]]; then
+            echo "::error:: Resource not found. Please check the organization ($ORG), team ($TEAMS), and username ($USER). $USER is not memeber of specified teams."
+            exit 1
+          fi
+
+          echo "$USER has the correct permissions to execute the workflow."
+
+      - name: Checkout repo @ current ref
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Extract deployment reusable workflow used tag
+        id: extract-deploy-wf-tag
+        run: |
+          set -euo pipefail
+
+          WORKFLOW_FILE=".github/workflows/deployment.yaml"
+
+          if [[ ! -f "$WORKFLOW_FILE" ]]; then
+            echo "::error:: Workflow file '$WORKFLOW_FILE' not found."
+            exit 1
+          fi
+          job_content=$(yq e '.jobs.deploy // ""' .github/workflows/deployment.yaml)
+          if [[ -z "$job_content" ]]; then
+            echo "::error:: 'deploy' job not found in the workflow file."
+            exit 1
+          fi
+
+          fullref=$(yq e '.jobs."deploy".uses' .github/workflows/deployment.yaml)
+
+          tag=${fullref#*@}
+
+          echo "fullref=$fullref"
+          echo "tag=$tag"
+
+          if [[ -n "${{ vars.INFRA_COMMONS_TAG }}" ]]; then
+            final_tag="${{ vars.INFRA_COMMONS_TAG }}"
+          else
+            final_tag="$tag"
+          fi
+          echo "infra_commons_tag_computed=$final_tag" >> $GITHUB_OUTPUT
+
+      - name: Show extracted tag
+        run: echo "infra_commons_tag_computed = ${{ steps.extract-deploy-wf-tag.outputs.infra_commons_tag_computed }}"
+
+  create_runner:
+    name: Create Self-Hosted Runner
+    runs-on: ubuntu-22.04
+    if: ${{ inputs.run_k8s_workflow || inputs.run_tf_apply_secrets_workflow }}
+    environment: ${{ inputs.environment }}
+    needs: [ initChecks ]
+    outputs:
+      ecs_task_id: ${{ steps.start_runner.outputs.ecs_task_id }}
+    strategy:
+      matrix:
+        index: [1,2,3,4,5,6,7,8,9,10]
```

```suggestion
        index: [1,2,3]
```
we only have 2 microservices and 1 job for now